File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcProfileDef/IfcParameterizedProfileDef/IfcTShapeProfileDef/IfcTShapeProfileDef.vb
Public Class IfcTShapeProfileDef Inherits IfcParameterizedProfileDef
    Public Depth As IfcPositiveLengthMeasure
    Public FlangeWidth As IfcPositiveLengthMeasure
    Public WebThickness As IfcPositiveLengthMeasure
    Public FlangeThickness As IfcPositiveLengthMeasure
    Public FilletRadius As IfcNonNegativeLengthMeasure
    Public FlangeEdgeRadius As IfcNonNegativeLengthMeasure
    Public WebEdgeRadius As IfcNonNegativeLengthMeasure
    Public WebSlope As IfcPlaneAngleMeasure
    Public FlangeSlope As IfcPlaneAngleMeasure

    ' === WHERE CLAUSES ===
    ' ValidFlangeThickness : FlangeThickness < Depth
    ' ValidWebThickness : WebThickness < FlangeWidth
End Class

File: output/ifc4x3_dev_73740fe4/entity/IfcTShapeProfileDef.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


public class IfcTShapeProfileDef extends IfcParameterizedProfileDef {

    public IfcPositiveLengthMeasure Depth;
    public IfcPositiveLengthMeasure FlangeWidth;
    public IfcPositiveLengthMeasure WebThickness;
    public IfcPositiveLengthMeasure FlangeThickness;
    public IfcNonNegativeLengthMeasure FilletRadius; // OPTIONAL
    public IfcNonNegativeLengthMeasure FlangeEdgeRadius; // OPTIONAL
    public IfcNonNegativeLengthMeasure WebEdgeRadius; // OPTIONAL
    public IfcPlaneAngleMeasure WebSlope; // OPTIONAL
    public IfcPlaneAngleMeasure FlangeSlope; // OPTIONAL
    // INVERSE attributes:
    // ValidFlangeThickness : FlangeThickness < Depth;
    // ValidWebThickness : WebThickness < FlangeWidth;

    // WHERE constraints:
    // ValidFlangeThickness : FlangeThickness < Depth;
    // ValidWebThickness : WebThickness < FlangeWidth;
}
